### ISSUE TITLE
Using stealing skill on scenery?

### ISSUE THREAD

**Issue by Lexx2k** [phobos2077/sfall#257]
The "Urn" scenery in Fo1 / Junktown seems hardcoded and allows that. In Fo2 however, this doesn't work anymore and using the stealing skill does nothing.

Is there a way to enable the use of the stealing skill on scenery objects? Setting some byte or something via set_proto_data maybe?

**Comment by FakelsHub** [comment]
Using a container instead of scenery does not solve the problem?

**Comment by Lexx2k** [comment]
Tried already. Problem with container is that it doesn't have the transparent egg, so if you stand behind the scenery, it sticks out very badly compared to the rest of the bar. You also can't set the east/west, etc. position flag on it (of course, as it doesn't have transparency in the first place). 

Also for some weird reason I wasn't able to cancel the inventory opening (script_overrides only worked the first time), but I didn't investigate much more into it right now.

**Comment by FakelsHub** [comment]
> Tried already. Problem with container is that it doesn't have the transparent egg, so if you stand behind the scenery, it sticks out very badly compared to the rest of the bar. 

Make the bar not transparent. I do not see a problem using the container.

**Comment by Lexx2k** [comment]
Making the bar not transparent means if Neal or anyone else dies behind it, you can't reach him.

**Comment by FakelsHub** [comment]
> Also for some weird reason I wasn't able to cancel the inventory opening 

Let it open, as it is an urn and something can be put into it)

**Comment by Lexx2k** [comment]
You do not open the urn, you open the bar piece - even if the urn is gone.

**Comment by FakelsHub** [comment]
> You do not open the urn, you open the bar piece - even if the urn is gone.

Do you have Photoshop is banned) to make a new object?

**Comment by FakelsHub** [comment]
> Making the bar not transparent means if Neal or anyone else dies behind it, you can't reach him.

you create problems out of nothing.

**Comment by Lexx2k** [comment]
I just don't like workarounds like this. It was merely a question if such a proto flag change is possible. Based on the arguing, I'm guessing no.

**Comment by FakelsHub** [comment]
>  It was merely a question if such a proto flag change is possible.

it's possible.

**Comment by burner1024** [comment]
Late to the party, but doesn't HOOK_USESKILL work for this?

**Comment by Lexx2k** [comment]
Nah, HOOK_USESKILL doesn't work, because the skill use isn't even triggered - you simply can't click the scenery with the stealing skill active.

**Comment by burner1024** [comment]
Hmm... how about HOOK_MOUSECLICK with obj_under_cursor + get_cursor_mode?

**Comment by Lexx2k** [comment]
I did some experiments, but I'm pretty much stuck. What works:
- detecting the object
- detecting cursor mode
- switching cursor mode back to 0 after clicking the object

Problems:
- Dude runs to the clicked hex (likely because of the cursor mode change to 0)
- I have no idea how to prevent that.
- I have no idea how to make the player walk to the object and play the use animation.

There is probably more, but that's the steps I'm stuck at.

**Comment by FakelsHub** [comment]
Ok. I will add this option for you. :)

**Comment by Lexx2k** [comment]
I was kinda hoping to use this also on other things - like, science and repair on other critters. Of course it has the same problems right now as what I wrote above. 
There is a ddraw.ini option to enable repair/science for all critters, but being able to do this only for special critters would be a lot better, imo.

**Comment by FakelsHub** [comment]
I tried to use the hook and I'm doing fine, there is only one problem-it is after the animation is used to call the theft handler.
There are solutions:
1 use the loop in the global script and catch the ending of the animation use
2. try using `add_timer_event({ObjectPtr obj}, {int time}, {int info});` for the dude and in his script handle the theft event.
3. in the next version of sfall it will be possible to use the event handler in global scripts it will suit best for this situation.
```
/*
   HOOK_USESKILLON (hs_useskillon.int)
   Critter arg1 - The user critter (usually dude_obj)
   Obj     arg2 - the target object/critter
   int     arg3 - skill used
   int     ret1 - a new critter who used the skill, pass -1 to cancel the skill use, 0 - for skip this return argument
   int     ret2 - pass 1 to allow the skill being used in combat (only for dude_obj or critter being controlled by the player)
*/

 variable
         a1 := get_sfall_arg,
         a2 := get_sfall_arg,
         a3 := get_sfall_arg;
   if (a3 == SKILL_STEAL) then begin
      if (obj_type(a2)  == OBJ_TYPE_SCENERY) then begin
         set_cursor_mode(0);
         reg_anim_begin();
         reg_anim_obj_run_to_obj(a1, a2, 0);
         reg_anim_animate(a1, ANIM_magic_hands_middle, 0);
         reg_anim_end();
      end
   end
```

**Comment by FakelsHub** [comment]
after studying the situation need to add new features:
`exec_obj_stdproc()`
`reg_anim_end_callback()`

**Comment by Lexx2k** [comment]
I tried your code and you're right, it works with HOOK_USESKILLON. 

I've added a timed_event_p_proc to the scenery script, imported a variable and call the event via my global script:
`if (Neal_urn_ptr != 0) then begin
   add_timer_event(Neal_urn_ptr, game_ticks(0), 1);
end`

This works, except that it gets called right away, like you mentioned. A callback for when "magic hands" was finished would be good.
